services/core-web/src/components/admin/complianceCodes/ComplianceReportManagement.tsx

```diff
@@ -1,204 +1,233 @@
-import { useAppDispatch as useDispatch, useAppSelector as useSelector } from "@mds/common/redux/rootState";
-import { ComplianceReportParams, fetchComplianceReports, getComplianceReportPageData, getReportDefinitionsLoaded } from "@mds/common/redux/slices/complianceReportsSlice";
-import { renderActionsColumn, renderTextColumn } from "@mds/common/components/common/CoreTableCommonColumns";
+import {
+  useAppDispatch as useDispatch,
+  useAppSelector as useSelector,
+} from "@mds/common/redux/rootState";
+import {
+  ComplianceReportParams,
+  createMineReportDefinition,
+  fetchComplianceReports,
+  fetchMineReportDueDateTypes,
+  getComplianceReportPageData,
+  getReportDefinitionsLoaded,
+} from "@mds/common/redux/slices/complianceReportsSlice";
+import {
+  renderActionsColumn,
+  renderTextColumn,
+} from "@mds/common/components/common/CoreTableCommonColumns";
 import { IComplianceArticle, IMineReportDefinition } from "@mds/common/interfaces";
-import { REPORT_REGULATORY_AUTHORITY_CODES, REPORT_REGULATORY_AUTHORITY_ENUM } from "@mds/common/constants/enums";
+import {
+  REPORT_REGULATORY_AUTHORITY_CODES,
+  REPORT_REGULATORY_AUTHORITY_ENUM,
+} from "@mds/common/constants/enums";
 import { formatComplianceCodeArticleNumber } from "@mds/common/redux/utils/helpers";
 import { EMPTY_FIELD, REPORT_REGULATORY_AUTHORITY_CODES_HASH } from "@mds/common/constants/strings";
 import { removeNullValues } from "@mds/common/constants/utils";
 import SearchOutlined from "@ant-design/icons/SearchOutlined";
-import { useHistory, useLocation } from "react-router-dom";
-import { ADMIN_HSRC_COMPLIANCE_CODE_MANAGEMENT } from "@/constants/routes";
-
+import { closeModal, openModal } from "@mds/common/redux/actions/modalActions";
+import AddReportDefinitionForm from "@/components/admin/complianceCodes/AddReportDefinitionForm";
 
 const defaultParams = {
-    page: 1,
-    per_page: 50
+  page: 1,
+  per_page: 50,
 };
 
 const ComplianceReportManagement: FC = () => {
-    const dispatch = useDispatch();
-    const { search } = useLocation();
-    const history = useHistory();
-    const initialParams = queryString.parse(search);
-    const reportPageData = useSelector(getComplianceReportPageData);
-    const reportDefinitions = reportPageData?.records ?? [];
-    const [sectionSearchText, setSectionSearchText] = useState(null);
-    const [queryParams, setQueryParams] = useState<ComplianceReportParams>({ ...defaultParams, ...initialParams });
-    const isLoaded = useSelector(getReportDefinitionsLoaded(queryParams));
-
-    const fetchData = () => {
-        dispatch(fetchComplianceReports(queryParams));
-    };
+  const dispatch = useDispatch();
+  const reportPageData = useSelector(getComplianceReportPageData);
+  const reportDefinitions = reportPageData?.records ?? [];
+  const [sectionSearchText, setSectionSearchText] = useState(null);
+  const [queryParams, setQueryParams] = useState<ComplianceReportParams>(defaultParams);
```

Comment:
Looks like there were a lot of issues with merge conflict resolution in this file. All the stuff with parsing the query from the url (useLocation, useHistory, initialParams)